pkg/upgrade/initialize.go

```diff
@@ -10,28 +10,31 @@ import (
 	"github.com/openshift/openshift-azure/pkg/util/azureclient/storage"
 )
 
-func (si *simpleUpgrader) InitializeCluster(ctx context.Context, cs *api.OpenShiftManagedCluster) error {
-	authorizer, err := azureclient.NewAuthorizerFromContext(ctx)
-	if err != nil {
-		return err
+func (u *simpleUpgrader) InitializeCluster(ctx context.Context, cs *api.OpenShiftManagedCluster) error {
+	if u.accountsClient == nil {
+		authorizer, err := azureclient.NewAuthorizerFromContext(ctx)
+		if err != nil {
+			return err
+		}
+
+		u.accountsClient = azureclient.NewAccountsClient(cs.Properties.AzProfile.SubscriptionID, authorizer, u.pluginConfig.AcceptLanguages)
 	}
-
-	accounts := azureclient.NewAccountsClient(cs.Properties.AzProfile.SubscriptionID, authorizer, si.pluginConfig.AcceptLanguages)
-	keys, err := accounts.ListKeys(ctx, cs.Properties.AzProfile.ResourceGroup, cs.Config.ConfigStorageAccount)
-	if err != nil {
-		return err
+	if u.storageClient == nil {
```

Comment:
I don't like this, but I guess it's because the SA doesn't exist when NewSimpleUpgrader is called.  I'm not immediately coming up with any better alternatives though - can you?
Certainly warrants a comment.

Reply:
unless you pass the context and cs into NewSimpleUpgrader

Reply:
opened follow up issue #602